internal/compiler/object_tree.rs

```diff
@@ -1980,6 +2135,134 @@ fn apply_default_type_properties(element: &mut Element) {
+        return;
+    }
+
+    for uses_identifier_node in uses_specifier.UsesIdenfifierList().UsesIdentifier() {
+        let Ok(uses_statement): Result<UsesStatement, ()> = (&uses_identifier_node).try_into()
+        else {
+            // We should already have reported a syntax error
+            continue;
+        };
+
+        let Ok(interface) = uses_statement.lookup_interface(tr, diag) else {
+            continue;
+        };
+
+        let Some(child) = find_element_by_id(e, &uses_statement.child_id) else {
+            diag.push_error(
+                format!("'{}' does not exist", uses_statement.child_id),
+                &uses_statement.child_id_node,
+            );
+            continue;
+        };
+
+        if !element_implements_interface(&child, &interface, &uses_statement, diag) {
+            continue;
+        }
+
+        for (prop_name, prop_decl) in &interface.root_element.borrow().property_declarations {
+            let lookup_result = e.borrow().base_type.lookup_property(prop_name);
+            if lookup_result.is_valid() {
+                diag.push_error(
+                    format!(
+                        "Cannot use interface '{}' because property '{}' conflicts with existing property in '{}'",
+                        uses_statement.interface_name, prop_name, e.borrow().base_type
+                    ),
+                    &uses_statement.interface_name_node,
+                );
+                continue;
+            }
+
+            if let Some(existing_property) =
+                e.borrow_mut().property_declarations.insert(prop_name.clone(), prop_decl.clone())
+            {
+                let source = existing_property
+                    .node
+                    .as_ref()
+                    .and_then(|node| node.child_node(SyntaxKind::DeclaredIdentifier))
+                    .and_then(|node| node.child_token(SyntaxKind::Identifier))
+                    .map_or_else(
+                        || parser::NodeOrToken::Node(uses_statement.child_id_node.clone().into()),
+                        |token| parser::NodeOrToken::Token(token),
+                    );
+
+                diag.push_error(
+                    format!(
+                        "Cannot override property '{}' from '{}'",
+                        prop_name, uses_statement.interface_name
+                    ),
+                    &source,
+                );
+                continue;
+            }
+
+            if let Some(existing_binding) = e.borrow_mut().bindings.insert(
+                prop_name.clone(),
+                BindingExpression::new_two_way(
+                    NamedReference::new(&child, prop_name.clone()).into(),
+                )
+                .into(),
+            ) {
+                let message = format!(
+                    "Cannot override binding for property '{}' from interface '{}'",
+                    prop_name, uses_statement.interface_name
+                );
```

Comment:
I struggled to come up with a test case for this error - the previous checks caught it first. Suggestions would be appreciated.